src/timeout.c

```diff
@@ -182,7 +189,13 @@ int getTimeoutFromObjectOrReply(client *c, robj *object, mstime_t *timeout, int
     }
 
     if (tval > 0) {
-        tval += mstime();
+        tval += now;
+        
+        /* if addition of `now` caused to overflow */
+        if (tval < 0) {
+            addReplyError(c,"timeout is out of range");
+            return C_ERR;
+        }
```

Comment:
```suggestion
        if (tval > LLONG_MAX - now) {
            addReplyError(c,"timeout is out of range");
            return C_ERR;
        }
        tval += now;
```

Signed integer overflow is undefined behavior and undefined behavior sanitizer will complain about this (If the test case is hitting here). So, I think we can use the above version to check it without overflow.

Reply:
Cool. thanks.